File: jdroid-android-core/src/main/java/com/jdroid/android/firebase/performance/FirebasePerformanceAppLifecycleCallback.java
package com.jdroid.android.firebase.performance;

import android.content.Context;

import com.google.firebase.perf.FirebasePerformance;
import com.jdroid.android.application.AbstractApplication;
import com.jdroid.android.lifecycle.ApplicationLifecycleCallback;

public class FirebasePerformanceAppLifecycleCallback extends ApplicationLifecycleCallback {

	@Override
	public void onProviderInit(Context context) {
		FirebasePerformance.getInstance().setPerformanceCollectionEnabled(FirebasePerformanceAppContext.isFirebasePerformanceEnabled());
		if (FirebasePerformanceAppContext.isFirebasePerformanceEnabled()) {
			AbstractApplication.get().registerActivityLifecycleCallbacks(FirebasePerformanceLifecycleCallbacks.get());
		}
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/firebase/performance/FirebasePerformanceAppLifecycleCallback.kt
package com.jdroid.android.firebase.performance

import android.content.Context

import com.google.firebase.perf.FirebasePerformance
import com.jdroid.android.application.AbstractApplication
import com.jdroid.android.lifecycle.ApplicationLifecycleCallback

class FirebasePerformanceAppLifecycleCallback : ApplicationLifecycleCallback() {

    override fun onProviderInit(context: Context) {
        FirebasePerformance.getInstance().isPerformanceCollectionEnabled = FirebasePerformanceAppContext.isFirebasePerformanceEnabled()
        if (FirebasePerformanceAppContext.isFirebasePerformanceEnabled()) {
            AbstractApplication.get().registerActivityLifecycleCallbacks(FirebasePerformanceLifecycleCallbacks.get())
        }
    }
}
